bee-common/bee-common/src/packable.rs

```diff
@@ -38,6 +39,73 @@ pub trait Packable {
         Self: Sized;
 }
 
+impl Packable for bool {
+    type Error = std::io::Error;
+
+    fn packed_len(&self) -> usize {
+        (*self as u8).packed_len()
+    }
+
+    fn pack<W: Write>(&self, writer: &mut W) -> Result<(), Self::Error> {
+        (*self as u8).pack(writer)
+    }
+
+    fn unpack<R: Read + ?Sized>(reader: &mut R) -> Result<Self, Self::Error>
+    where
+        Self: Sized,
+    {
+        Ok(match u8::unpack(reader)? {
+            0 => false,
+            _ => true,
+        })
+    }
+}
+
+/// Error that occurs on `Option<P: Packable>` operations.
+#[derive(Debug)]
+pub enum OptionError<P> {
+    /// Error that occurs on boolean `Packable` operations.
+    Bool(<bool as Packable>::Error),
+    /// Error that occurs on inner `Packable` operations.
+    Inner(P),
+}
+
+impl<P: Packable> Packable for Option<P> {
+    type Error = OptionError<P::Error>;
+
+    fn packed_len(&self) -> usize {
+        true.packed_len()
+            + match self {
+                Some(p) => p.packed_len(),
+                None => 0,
+            }
+    }
+
+    fn pack<W: Write>(&self, writer: &mut W) -> Result<(), Self::Error> {
+        match self {
+            Some(p) => {
+                true.pack(writer).map_err(|e| OptionError::Bool(e))?;
```

Comment:
You can just do `.map_err(OptionError::Bool)`

Tuple enum variants (and tuple structs) are implicitly just functions